server/utilities/smart-turn.mdx

```diff
@@ -0,0 +1,248 @@
+        ),
+    ),
+)
+```
+
+## Local Smart Turn (CoreML)
+
+The `LocalCoreMLSmartTurnAnalyzer` runs inference locally using CoreML, providing lower latency and no network dependencies.
+
+### Constructor Parameters
+
+<ParamField path="smart_turn_model_path" type="str" required>
+  Path to the directory containing the Smart Turn model
+</ParamField>
+
+<ParamField path="sample_rate" type="Optional[int]" default="None">
+  Audio sample rate (will be set by the transport if not provided)
+</ParamField>
+
+<ParamField path="params" type="SmartTurnParams" default="SmartTurnParams()">
+  Configuration parameters for turn detection
+</ParamField>
+
+### Example
+
+```python
+import os
+from pipecat.audio.turn.local_smart_turn import LocalCoreMLSmartTurnAnalyzer
+from pipecat.audio.turn.base_smart_turn import SmartTurnParams
+from pipecat.transports.base_transport import TransportParams
+
+# Path to the Smart Turn model directory
+smart_turn_model_path = os.getenv("LOCAL_SMART_TURN_MODEL_PATH")
+
+# Create the transport with local Smart Turn detection
+transport = SmallWebRTCTransport(
+    webrtc_connection=webrtc_connection,
+    params=TransportParams(
+        audio_in_enabled=True,
+        audio_out_enabled=True,
+        vad_enabled=True,
+        vad_analyzer=SileroVADAnalyzer(params=VADParams(stop_secs=0.2)),
+        vad_audio_passthrough=True,
+        end_of_turn_analyzer=LocalCoreMLSmartTurnAnalyzer(
+            smart_turn_model_path=smart_turn_model_path,
+            params=SmartTurnParams(
+                stop_secs=2.0,  # Shorter stop time when using Smart Turn
+                pre_speech_ms=0.0,
+                max_duration_secs=8.0
+            )
+        ),
+    ),
+)
+```
+
+## Local Model Setup
+
+To use the `LocalCoreMLSmartTurnAnalyzer`, you need to set up the CoreML model locally. Follow these steps from the [official repository instructions](https://github.com/pipecat-ai/smart-turn/blob/main/coreml/README.md):
+
+1. Install Git LFS (Large File Storage):
+
+   ```bash
+   # On macOS
+   brew install git-lfs
+
+   # On Ubuntu/Debian
+   sudo apt-get install git-lfs
+
+   # Initialize Git LFS
+   git lfs install
+   ```
+
+2. Clone the Smart Turn model repository:
+
+   ```bash
+   git clone https://huggingface.co/pipecat-ai/smart-turn
+   ```
+
+3. Set the environment variable to the cloned repository path:
+
+   ```bash
+   # Add to your .env file or environment
+   export LOCAL_SMART_TURN_MODEL_PATH=/path/to/smart-turn
+   ```
+
+4. Verify the CoreML model exists:
+   ```bash
+   # The CoreML model should be located at:
+   ls /path/to/smart-turn/coreml/smart_turn_classifier.mlpackage
+   ```
+
+Note that the CoreML model is optimized for Apple Silicon devices. If you're using a different platform, the remote Smart Turn service might be a better option.
+
+## How It Works
+
+Smart Turn Detection works in three main phases:
+
+1. **Audio Collection**: When the user starts speaking, the system begins collecting audio.
```

Comment:
The system is always collecting audio, or silence, if the user is not speaking. 
When silence is detected, we maintain a small buffer to fulfill the `pre_speech_ms` setting.